tf-modules/cumulus/ecs_cluster_instance_autoscaling_cf_template.yml.tmpl

```diff
@@ -95,10 +95,18 @@ Resources:
       AutoScalingRollingUpdate:
         MinInstancesInService: ${min_size}
     Properties:
+%{ if length(availability_zones) > 0 ~}
+      AvailabilityZones:
```

Comment:
Is there a time that we would want to create an autoscaling group without specifying the subnets that it should use? From my reading of the AWS docs, it looks like `AvailabilityZones` is only required for EC2-Classic instances.

* https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-as-group.html#cfn-as-group-availabilityzones
* https://www.terraform.io/docs/providers/aws/r/autoscaling_group.html

Reply:
This change was required because I got an error when neither `AvailabilityZones` nor `VPCZoneIdentifier` were supplied. The error indicated that one or the other must be specified.

Reply:
The real issue here was that since Chuck had no `subnet_ids` set, the `VPCZoneIdentifier` block was empty and deploying this ECS cluster was failing. As the docs say:

> A list of Availability Zones for the group. You must specify one of the following properties: VPCZoneIdentifier or AvailabilityZones. 

So dding the ability to set the availability zones was just a workaround.

@yjpa7145 - Is there a better solution here? Is there any way to make the whole ECS cluster optional